test/OpenTelemetry.Tests/Instrumentation/PropertyFetcherTest.cs

```diff
@@ -78,6 +78,24 @@ public void FetchPropertyMultiplePayloadTypes_IgnoreTypesWithoutExpectedProperty
         Assert.Equal("A", propertyValue);
     }
 
+    [Fact]
```

Comment:
Possible gaps in tests:

* A test to verify the throw behavior when a root value type is handed to property fetcher.
* Does one of these tests fall into the `if (declaringType == typeof(object))` branch? Something like this (I think 😄)...

   ```csharp
      private class PayloadTypeWithObjectType
       {
           public object Property { get; set; } = new DerivedType();
       }
   ```

Reply:
> * A test to verify the throw behavior when a root value type is handed to property fetcher.

-> this was resolved in https://github.com/open-telemetry/opentelemetry-dotnet/pull/4675/commits/cabf49f5f6e87a2e401b0fdd72621d10d9e1ffb7

> * Does one of these tests fall into the `if (declaringType == typeof(object))` branch? Something like this (I think 😄)...
>   ```cs
>      private class PayloadTypeWithObjectType
>       {
>           public object Property { get; set; } = new DerivedType();
>       }
>   ```
-> Check out this discussion: 
https://github.com/open-telemetry/opentelemetry-dotnet/pull/4675#discussion_r1290495522
and the updated comment on this specific if branch: https://github.com/open-telemetry/opentelemetry-dotnet/pull/4675/commits/b7811147e5f9b5da3a77bcad38207601a5ccfed0